### ISSUE TITLE
REFL016 should respect 'this'-qualification style rules

### ISSUE THREAD

**Issue by jnm2** [DotNetAnalyzers/ReflectionAnalyzers#130]
Editorconfig:

```ini
[*]
dotnet_style_qualification_for_event = false:suggestion
dotnet_style_qualification_for_field = false:suggestion
dotnet_style_qualification_for_method = false:suggestion
dotnet_style_qualification_for_property = false:suggestion
```

```c#
class C
{
    void M()
    {
        typeof(C).GetMethod(nameof(this.M));
    }
}
```

> REFL016 Use nameof(this.M).

```c#
typeof(C).GetMethod(nameof(this.M));
```

Expected:

> REFL016 Use nameof(M).

```c#
typeof(C).GetMethod(nameof(M));
```

This might be a Roslyn bug, because `nameof(this.M)` does not trigger the IDE003 diagnostic even though `new Action(this.M)` and `this.M()` do.

**Comment by JohanLarsson** [comment]
Figuring out where `.editorconfig` is and reading it should not be too hard but may be a bit slow, dunno.

**Comment by jnm2** [comment]
Surely Roslyn ancapsulates the actual reading of the .editorconfig file hierarchy and merges it with the user's current C# options?

**Comment by JohanLarsson** [comment]
Dunno, @sharwell probably knows this, I think I have asked in the past but don't remember the answer and things may have changed.

**Comment by brian-reichle** [comment]
> Surely Roslyn encapsulates the actual reading of the .editorconfig file hierarchy and merges it with the user's current C# options?

They certainly seem to.

https://github.com/dotnet/roslyn/blob/cbcc9ecce81aeb1ed85c7ddd6fd0e9b199b913e6/src/Workspaces/Core/Portable/CodeStyle/CodeStyleOptions.cs#L40-L43
